src/PromiseCore.php

```diff
@@ -192,7 +192,12 @@ public function reject(Exception $exception)
     {
         $this->exception = $exception;
         $this->state = Promise::REJECTED;
-        $this->exception = $this->call($this->onRejected, $this->exception);
+
+        try {
+            $this->call($this->onRejected, $this->exception);
+        } catch (Exception $exception) {
+            $this->exception = $exception;
```

Comment:
This will replace exception related to request with exception related to callback code. Why?